src/site/antora/modules/ROOT/pages/manual/webapp.adoc

```diff
@@ -15,311 +15,626 @@
+Jetty::
+Recent versions of Jetty have a `logging-log4j2` module that can be easily enabled to share Log4j Core between applications and to use Log4j Core for the Jetty server itself.
+See {jetty-latest-url}/operations-guide/modules/index.html[Jetty Modules documentation] for more details.
 
-Thankfully, Log4j provides a simple mechanism for binding the LoggerContext to asynchronous threads in these special circumstances. The simplest way to do this is to wrap the Runnable instance that is passed to the AsyncContext.start() method.
+OpenLiberty::
+In OpenLiberty, you can add Log4j as a _global library_.
+See https://openliberty.io/docs/latest/class-loader-library-config.html#_configure_global_libraries[OpenLiberty documentation] for more details.
 
-[source,java]
-----
-import java.io.IOException;
-import javax.servlet.AsyncContext;
-import javax.servlet.ServletException;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
+Payara::
+See https://docs.payara.fish/community/docs/Technical%20Documentation/Payara%20Server%20Documentation/Server%20Configuration%20And%20Management/Classloading/Standard%20Classloading.html#common-libraries[Payara Common Libraries documentation].
+
+Tomcat::
+In Tomcat, you can use the _common classloader_.
+See {tomcat-latest-url}/class-loader-howto.html[Tomcat classloader documentation] for more details.
+
+WildFly::
+You can install Log4j as a _global module_ or in a _global directory_.
+See {wildfly-latest-url}/Admin_Guide.html#EE_Application_Deployment_Configuration[WildFly EE Application Deployment documentation] for more details.
++
+Check also the <<web-application-wildfly,WildFly note above>>.
+
+[WARNING]
+====
+Web application classloaders (see
+https://jakarta.ee/specifications/servlet/6.0/jakarta-servlet-spec-6.0#web-application-class-loader[Servlet Specification 10.7.2]
+) use a _"parent last"_ delegation strategy, but prevent application from overriding implementation classes provided by the container.
+
+If you share Log4j between applications and the applications themselves contain Log4j Core, the logging behavior depends on the application server.
+Some application servers will use the shared instance (e.g. WildFly), while others will use the application instance (e.g. Tomcat).
+There are two solutions to this problem:
+
+* you can remove Log4j from the WAR or EAR archive:
+
+Maven::
+If you use Maven, you can declare the
+https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Scope[scope]
+of all Log4j libraries as `provided`.
+
+Gradle::
+If you use Gradle, you can add `log4j-api` to the `providedCompile` configuration, while `log4j-core` to the `providedRuntime` configuration.
+See https://docs.gradle.org/current/userguide/war_plugin.html#sec:war_dependency_management[Gradle WAR plugin] for more details.
```

Comment:
Mind placing these in tab block, please? :innocent: That will comply with the convention followed in the rest of the manual.

Reply:
Fixed in 2809eaae3db7089f9c56b9b1e5a218e0606a4f38.